fl4health/utils/losses.py

```diff
@@ -128,26 +152,36 @@ def aggregate_loss_by_type(
             loss_meter_type (LossMeterType): The type of LossMeter (ie average or accumulation).
 
         Returns:
-            Dict[str, torch.Tensor]: The dictionary of aggregated losses.
+            Tuple[Dict[str, torch.Tensor], Dict[str, torch.Tensor]]: Two dictionaries with the aggregated losses.
+                The first one is the additional losses and the second one is the backward losses (or None if the losses
+                are not a subclass of TrainLosses)
         """
-
-        if loss_type == LossType.BACKWARD:
-            loss_list = [losses.backward for losses in losses_list]
-        else:
-            loss_list = [losses.additional_losses for losses in losses_list]
-
-        # We don't know the keys of the dict (backward or additional losses) beforehand. We obtain them
-        # from the first entry because we know all of the losses will have the same keys
-        loss_keys = loss_list[0].keys()
-        num_losses = len(loss_list)
-        loss_dict: Dict[str, torch.Tensor] = {}
-        for key in loss_keys:
-            loss = torch.sum(torch.FloatTensor([loss[key] for loss in loss_list]))
-            if loss_meter_type == LossMeterType.AVERAGE:
-                loss = loss / num_losses
-            loss_dict[key] = loss
-
-        return loss_dict
+        additional_loss_list = [losses.additional_losses for losses in losses_list]
+        backward_loss_list = None
+        # Assuming the losses are all the same type, check for the type of the first one only
+        if len(losses_list) > 0 and isinstance(losses_list[0], TrainLosses):
+            backward_loss_list = [losses.backward for losses in losses_list]  # type: ignore
```

Comment:
Perhaps for each of the Losses classes we can have a class function that takes all of the losses present and rolls them into lists that can all be processed by this aggregation and then "reconstituted from the lists thereafter. That way we could just call these class functions to deconstruct, aggregate, and reconstruct by just calling something like Losses.deconstruct(...) and Losses.reconstruct(...).

Reply:
We can talk about this in a bit more detail if you'd like.

Reply:
That makes a lot of sense.